runtime/onert/backend/cpu/SharedMemoryOperands.cc

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "SharedMemoryOperands.h"
+
+namespace onert
+{
+namespace backend
+{
+namespace cpu
+{
+
+namespace
+{
+// To handle cases like Reshape->Reshape->Reshape... chain where the memory is shared.
+// In such a case we should re-assign indexes to the first Reshape input.
+void reassign_indexes_to_single_sources(
+  ir::OperandIndexMap<ir::OperandIndex> &shared_memory_operand_map)
+{
+  for (auto [shared_ind, source_ind] : shared_memory_operand_map)
+  {
+    bool other_source_found = false;
+    auto it = std::end(shared_memory_operand_map);
+    while ((it = shared_memory_operand_map.find(source_ind)) != std::end(shared_memory_operand_map))
+    {
+      source_ind = shared_memory_operand_map[source_ind];
+      other_source_found = true;
+    }
+    if (other_source_found)
+    {
+      shared_memory_operand_map[shared_ind] = source_ind;
+    }
+  }
```

Comment:
If `shared_memory_operand_map = {{1, 2}, {2, 3}, {2, 4}}` is given, is this works well..?

Reply:
As I understand `2` cannot indicate other operand two times. (I've made some experiments about this implementation here - https://godbolt.org/z/11KKr9xqY)

Reply:
Oh I see. I misunderstood the details. 
Thank you for your implementations :)

My original question was that I it is safe with below graphs :

```
		   -  reshape - output operand  
		/ 
operand -  
		\
		  - other operation - output operand 
```

Reply:
ohh, I see, definitely it makes sense to write such unit test, thank you!

Reply:
I added such a test:
![image](https://github.com/user-attachments/assets/c2d0b40e-e947-40fd-81d1-2cd830f54599)
and it seems to work correctly (the memory is sharing)